dev/com.ibm.ws.security.oauth/test/com/ibm/ws/security/oauth20/plugins/OidcBaseClientValidatorTest.java

```diff
@@ -0,0 +1,338 @@
+        OidcBaseClientValidator validator = OidcBaseClientValidator.getInstance(client);
+        try {
+            validator.validateCreateUpdate();
+            fail("Did not throw OIDCServerException as expected!");
+        } catch (OidcServerException e) {
+            assertEquals(HttpServletResponse.SC_BAD_REQUEST, e.getHttpStatus());
+            assertEquals(OIDCConstants.ERROR_INVALID_CLIENT_METADATA, e.getErrorCode());
+            String expectedErrorMessage = "CWWKS1444E: The client registration metadata field response_type contains value id_token token, which requires at least a matching grant_type value implicit.";
+            assertEquals(expectedErrorMessage, e.getErrorDescription());
+        }
+    }
+
+    @Test
+    public void testMalformedRedirectURIs() throws OidcServerException
+    {
+        JsonArray redirectUris = new JsonArray();
+        redirectUris.add(new JsonPrimitive("http://finance.yahoo.com/q/h?s=^IXIC"));//invalid URI
+        client.setRedirectUris(redirectUris);
+        OidcBaseClientValidator validator = OidcBaseClientValidator.getInstance(client);
+        try {
+            validator.validateCreateUpdate();
+            fail("Did not throw OIDCServerException as expected!");
+        } catch (OidcServerException e) {
+            assertEquals(HttpServletResponse.SC_BAD_REQUEST, e.getHttpStatus());
+            assertEquals(OIDCConstants.ERROR_INVALID_REDIRECT_URI, e.getErrorCode());
+            String expectedErrorMessage = "CWWKS1445E: The value http://finance.yahoo.com/q/h?s=^IXIC for the client registration metadata field redirect_uris contains a malformed URI syntax.";
+            assertEquals(expectedErrorMessage, e.getErrorDescription());
+        }
+    }
+
+    @Test
+    public void testNonAbsoluteRedirectURIs() throws OidcServerException
+    {
+        JsonArray redirectUris = new JsonArray();
+        redirectUris.add(new JsonPrimitive("../finance.yahoo.com/q/h?s=time"));//invalid URI
+        client.setRedirectUris(redirectUris);
+        client.setApplicationType("web");
+        OidcBaseClientValidator validator = OidcBaseClientValidator.getInstance(client);
+        try {
+            validator.validateCreateUpdate();
+            fail("Did not throw OIDCServerException as expected!");
+        } catch (OidcServerException e) {
+            assertEquals(HttpServletResponse.SC_BAD_REQUEST, e.getHttpStatus());
+            assertEquals(OIDCConstants.ERROR_INVALID_REDIRECT_URI, e.getErrorCode());
+            String expectedErrorMessage = "CWWKS1446E: The value ../finance.yahoo.com/q/h?s=time for the client registration metadata field redirect_uris is not an absolute URI.";
+            assertEquals(expectedErrorMessage, e.getErrorDescription());
+        }
+    }
+
+    @Test
+    public void testDuplicateRedirectURIs() throws OidcServerException
+    {
+        JsonArray redirectUris = new JsonArray();
+        redirectUris.add(new JsonPrimitive("http://finance.yahoo.com/q/h"));
```

Comment:
Do we want to leave this as finance.yahoo? I know it's intentionally  a bad url...not sure if we want an IBM related bad url.

Reply:
Changing it to something IBM related would be better, yes. I've no idea why that URL was chosen.